File: 56-merge-intervals/merge-intervals.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        if(intervals.Length == 0) return new int[0][];

        Array.Sort(intervals, (a, b) => a[0].CompareTo(b[0]));

        List<int[]> merged = new List<int[]>();

        merged.Add(intervals[0]);
        foreach(var interval in intervals)
        {
           int[] last = merged.Last();

            if (interval[0] <= last[1]) {
                last[1] = Math.Max(last[1], interval[1]);
            } else {
                merged.Add(interval);
            }
        }

        return merged.ToArray();
    }
}

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n = intervals.size();
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> res;
        res.push_back(intervals[0]);
        int i = 1;
        while(i < n)
        {
            if(res.back()[1] >= intervals[i][0])
            {
                res.back()[1] = max(res.back()[1], intervals[i][1]);;
            }
            else
            {
                res.push_back(intervals[i]);
            }
            i++;
        }

        return res;
    }
};